distributed/protocol/serialize.py

```diff
@@ -195,6 +202,14 @@ def deserialize(self):
         frames = decompress(self.header, self.frames)
         return deserialize(self.header, frames)
 
+    def __eq__(self, other):
+        return (isinstance(other, Serialized) and
+                other.header == self.header and
+                other.frames == self.frames)
+
+    def __ne__(self, other):
+        return not (self == other)
```

Comment:
I don't have a specific example concern in this case, I've just been bitten by projects in the past where people define `__eq__` for convenience and then suddenly I can't use those objects with other python containers.  I now avoid `__eq__` by a strong habit unless there is a good API reason to implement it (such as with dask.array).  So this may just be an irrational avoidance on my part, but it's still something I'd prefer to avoid if it's not particularly costly to avoid having an `__eq__` method.